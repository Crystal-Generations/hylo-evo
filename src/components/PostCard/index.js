import component from './PostCard'
import connector from './PostCard.connector'
export default connector(component)
export { PostHeader, PostImage, PostBody, PostFooter } from './PostCard'
