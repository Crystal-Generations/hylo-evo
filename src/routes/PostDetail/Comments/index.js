import { withResizeDetector } from 'react-resize-detector'
import component from './Comments'
import connector from './Comments.connector'

export default connector(withResizeDetector(component, { handleHeight: false }))
