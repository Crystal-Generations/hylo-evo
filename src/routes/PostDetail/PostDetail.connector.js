import { connect } from 'react-redux'
import { get, find } from 'lodash/fp'
import { push } from 'react-router-redux'
import fetchPost from 'store/actions/fetchPost'
import getParam from 'store/selectors/getParam'
import getPost, { presentPost } from 'store/selectors/getPost'
import getMe from 'store/selectors/getMe'
import getCommunityForCurrentRoute from 'store/selectors/getCommunityForCurrentRoute'
import joinProject from 'store/actions/joinProject'
import leaveProject from 'store/actions/leaveProject'
import { FETCH_POST } from 'store/constants'

export function mapStateToProps (state, props) {
  const id = getParam('postId', state, props)
  const currentCommunity = getCommunityForCurrentRoute(state, props)
  const post = presentPost(getPost(state, props), get('id', currentCommunity))
  const slug = getParam('slug', state, props)
  const currentUser = getMe(state)
  const isProjectMember = find(({id}) => id === currentUser.id, get('members', post))

  return {
    id,
    post,
    currentUser,
    slug,
    isProjectMember,
    pending: state.pending[FETCH_POST]
  }
}

export const mapDispatchToProps = (dispatch, props) => {
  const { location } = props

  const removePostDetailFromPath = pathname =>
    pathname.replace(/\/p\/(.+)/, '')

  const closeLocation = {
    ...props.location,
    pathname: removePostDetailFromPath(location.pathname)
  }

  const postId = getParam('postId', {}, props)

  return {
    fetchPost: () => dispatch(fetchPost(postId)),
    onClose: () => dispatch(push(closeLocation)),
    editPost: () => dispatch(push(`${postId}/edit`)),
    joinProject: () => dispatch(joinProject(postId)),
    leaveProject: () => dispatch(leaveProject(postId))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
