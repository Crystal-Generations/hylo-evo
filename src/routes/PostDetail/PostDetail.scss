.post {
  padding-top: $space-6x;
  background-color: white;
}

.header {
  margin-bottom: 10px;
}

.header-sticky {
  position: fixed;
  top: $topnav-height;
  background-image: linear-gradient(0deg, rgba(255,255,255,0.00) 0%, $color-white 25%);
  padding: $space-6x 0;
  z-index: 1;
}

.activity-sticky {
  z-index: 1;
  position: fixed;
  top: $topnav-height;
  background: $color-white;
}

.image {
  margin-bottom: 10px;
}

.body {
  padding-left: 10px;
  padding-right: 10px;
}

.activity-header {
  // TODO: Extract Typography
  padding: 0 25px;
  margin-bottom: 20px;
  @include font-regular;
  color: $color-rhino-80;
  font-size: 17px;
}

.not-found {
  text-align: center;
  font-size: larger;
  padding: 50px;
}

// PostTag

.tags {
  padding: 0px 25px;
  margin-bottom: 10px;
}

.tag {
  // TODO: Extract typography
  @include font-regular;
  color: $color-caribbean-green;
  font-size: 12px;
  &:hover {
    text-decoration: none;
    color: $color-gossamer;
  }
  margin-right: 7px;
}
