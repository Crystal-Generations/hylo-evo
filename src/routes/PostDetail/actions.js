import { FETCH_POST } from 'store/constants'

export function fetchPost (id, opts = {}) {
  return {
    type: FETCH_POST,
    graphql: {
      query: `query ($id: ID) {
        post(id: $id) {
          id
          title
          details
          type
          creator {
            id
            name
            avatarUrl
          }
          createdAt
          updatedAt
          commenters(first: 2) {
            id
            name
            avatarUrl
          }
          commentersTotal
          linkPreview {
            title
            url
            imageUrl
          }
          votesTotal
          communities {
            id
            name
            slug
          }
        }
      }`,
      variables: {
        id
      }
    },
    meta: {
      rootModelName: 'Post'
    }
  }
}
