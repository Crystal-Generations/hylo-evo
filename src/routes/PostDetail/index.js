import component from './PostDetail'
import connector from './PostDetail.connector'

export default connector(component)
