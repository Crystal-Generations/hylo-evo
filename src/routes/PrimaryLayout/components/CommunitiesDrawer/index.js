import component from './CommunitiesDrawer'
import connector from './CommunitiesDrawer.connector'

export default connector(component)
