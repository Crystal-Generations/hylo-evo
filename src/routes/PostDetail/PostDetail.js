import React, { PropTypes, Component } from 'react'
import './PostDetail.scss'
const { object } = PropTypes
import { PostHeader, PostFooter } from 'components/PostCard/component'

export default class PostDetail extends Component {
  static propTypes = {
    post: object,
    match: object
  }

  render () {
    const { post, match: { params: { postId } } } = this.props
    return <div styleName='post'>
      <PostHeader creator={post.creator}
        date={post.updatedAt || post.createdAt}
        type={post.type}
        context={post.context}
        communities={post.communities} />
      <div>PostId: {postId}</div>
      <PostFooter id={post.id}
        commenters={post.commenters}
        commentersTotal={post.commentersTotal}
        votesTotal={post.votesTotal} />
    </div>
  }
}
