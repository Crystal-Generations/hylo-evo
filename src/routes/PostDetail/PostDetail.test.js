import PostDetail, { PostTags } from './PostDetail'
import { shallow } from 'enzyme'
import React from 'react'
import { PostImage, PostBody, PostFooter } from 'components/PostCard'
import PostHeader from 'components/PostCard/PostHeader'

describe('PostDetail', () => {
  it('renders correctly', () => {
    const imageUrl = 'foo.jpg'
    const tags = ['singing', 'dancing']
    const details = 'the body of the post'
    const votesTotal = 7
    const slug = 'foo'
    const myVote = true

    const post = {
      id: '91',
      imageUrl,
      tags,
      details,
      votesTotal,
      myVote
    }

    const wrapper = shallow(<PostDetail post={post} slug={slug} showCommunity />)
    expect(wrapper).toMatchSnapshot()
  })
})
