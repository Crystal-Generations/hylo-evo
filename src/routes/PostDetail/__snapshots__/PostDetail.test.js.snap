// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetail renders correctly 1`] = `
<ResizeDetector
  handleHeight={false}
  handleWidth={true}
  nodeType="div"
  onResize={[Function]}
  querySelector={null}
  refreshRate={1000}
  skipOnMount={false}
  targetDomEl={null}
  targetRef={null}
>
  <Component />
</ResizeDetector>
`;
